backbone.nativeajax.js

```diff
@@ -44,16 +45,33 @@
 
         var status = xhr.status;
         var data = getData(options.headers && options.headers.Accept, xhr);
+        var aCallbackFailed = false;
+        var reason;
 
         // Check for validity.
         if (isValid(xhr)) {
-          if (options.success) options.success(data);
-          if (resolve) resolve(data);
+          try {
+            options.success && options.success(data);
+          } catch(e) {
+            reason = e;
```

Comment:
what i've come across is when the success or error callbacks is synchronous, line 50 in the original fails and then resolve/reject never happen. i don't see a promise catch handler that could handle an error in success or error